src/main/java/com/greenfox/poker/service/TokenService.java

```diff
@@ -0,0 +1,81 @@
+package com.greenfox.poker.service;
+
+import com.greenfox.poker.model.PokerUser;
+
+import com.greenfox.poker.repository.PokerUserRepo;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MissingClaimException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.impl.crypto.MacProvider;
+
+import java.security.Key;
+import java.util.*;
+import java.lang.*;
+import javax.validation.constraints.Null;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TokenService {
+
+  @Autowired
+  PokerUserRepo pokerUserRepo;
+
+  private Key key;
+
+  public String generateToken(PokerUser pokerUser) {
+    HashMap<String, Object> claims = new HashMap<>();
+    claims.put("id", pokerUser.getId());
+    claims.put("username", pokerUser.getUsername());
+    String token = generateTokenByClaims(claims);
+    return token;
+  }
+
+  private String generateTokenByClaims(HashMap<String, Object> claims) {
+    this.key = MacProvider.generateKey();
+    String token = Jwts.builder()
+        .setClaims(claims)
+        .signWith(SignatureAlgorithm.HS256, key)
+        .compact();
+    return token;
+  }
+
+  private Claims getClaimsFromToken(String token) {
+    Claims claims;
+    claims = Jwts.parser()
+        .setSigningKey(key)
+        .parseClaimsJws(token)
+        .getBody();
+    return claims;
+  }
+
+  public String getUsernameFromToken(String token) {
+    String username;
+    try {
+      Claims claims = getClaimsFromToken(token);
+      username = (String) claims.get("username");
+    } catch (MissingClaimException e) {
+      username = null;
+    }
+    return username;
+  }
+
+  public long getIdFromToken(String token) {
+    long id;
+    try {
+      Claims claims = getClaimsFromToken(token);
+      id = (Integer) claims.get("id");
+    } catch (MissingClaimException e) {
+      id = -1L;
+    }
+    return id;
+  }
```

Comment:
`getIdFromToken(String token) ` is being used inside the `getPokerUserFromToken(String token)`.
But You maybe right about `getUsernameFromToken(String token)`